include/seqan3/range/container/dynamic_bitset.hpp

```diff
@@ -0,0 +1,1832 @@
+    constexpr dynamic_bitset(uint64_t const value)
+    {
+        if (detail::popcount(value >> 58) != 0)
+            throw std::invalid_argument{"The dynamic_bitset can be at most 58 long."};
+        data_.bits |= value;
+        data_.size |= value ? detail::most_significant_bit_set(value) + 1 : 0u;
+    }
+
+    /*!\brief Construct from two iterators.
+     * \tparam begin_it_type Must model std::ForwardIterator and `value_type` must be constructible from
+     *                       the reference type of begin_it_type.
+     * \tparam   end_it_type Must model std::Sentinel.
+     * \param[in]         begin_it Begin of range to construct/assign from.
+     * \param[in]           end_it End of range to construct/assign from.
+     *
+     * ### Complexity
+     *
+     * Linear in the distance between `begin_it` and `end_it`.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    template <std::ForwardIterator begin_it_type, std::Sentinel<begin_it_type> end_it_type>
+    //!\cond
+        requires std::Constructible<value_type, /*ranges::iter_reference_t*/reference_t<begin_it_type>>
+    //!\endcond
+    constexpr dynamic_bitset(begin_it_type begin_it, end_it_type end_it) noexcept:
+        dynamic_bitset{}
+    {
+        assign(begin_it, end_it);
+    }
+
+    /*!\brief Construct from a different range.
+     * \tparam other_range_t The type of range to be inserted; must satisfy std::ranges::InputRange and `value_type`
+     *                       must be constructible from reference_t<other_range_t>.
+     * \param[in]      range The sequence to construct/assign from.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of `range`.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    template <std::ranges::InputRange other_range_t>
+    //!\cond
+        requires !std::is_same_v<remove_cvref_t<other_range_t>, dynamic_bitset>
+                 /*ICE: && std::Constructible<value_type, reference_t<other_range_t>>*/
+    //!\endcond
+    explicit constexpr dynamic_bitset(other_range_t && range) noexcept :
+        dynamic_bitset{std::ranges::begin(range), std::ranges::end(range)}
+    {}
+
+    /*!\brief Construct with `n` times `value`.
+     * \param[in] n     Number of elements.
+     * \param[in] value The initial value to be assigned.
+     *
+     * ### Complexity
+     *
+     * Linear in `n`.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr dynamic_bitset(size_type n, value_type value) noexcept :
+        dynamic_bitset{}
+    {
+        assign(n, value);
+    }
+
+    /*!\brief Assign from `std::initializer_list`.
+     * \param[in] ilist A `std::initializer_list` of value_type.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of `ilist`.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr dynamic_bitset & operator=(std::initializer_list<value_type> ilist) noexcept
```

Comment:
The might of the ctor with a `initializer_list` overwhelms all other constructors regardless of constraints. Hence it is always chosen, but never works.
We also don't have this as ctor in other containers because of this reason. But maybe there is a way?